lib/macgyver.js

```diff
@@ -1,11 +1,88 @@
 var pristineEnv = require('./pristine-env');
 
+var pristineObject = pristineEnv().Object;
+var pristineArray = pristineEnv().Array;
+
+
+/*
+ * Copy an existing property from `from` to `to`.
+ * We don't want to pollute native prototypes with undefined properties,
+ * therefore we must check they exist first.
+ */
+function copy(to, from, key) {
+    if (from.hasOwnProperty(key)) {
+        to[key] = from[key];
+    }
+}
+
+/**
+ * Try to return a thing which prototype is in a pristine condition.
+ * Currently only supports arrays.
+ * If a thing cannot be fixed, the function returns the thing.
+ * The current native objects methods are not touched.
+ *
+ * @example
+ *     Array.prototype.map = function () {
+ *         //custom implementation
+ *     };
+ *
+ *    var arr = [1,2,3];
+ *    arr.map.toString();
+ *    //=> custom implementation
+ *    macgyver(arr).map.toString();
+ *    //=> native implementation
+ */
 function macgyver(thing) {
-    if (Object.prototype.toString.call(thing) === '[object Array]') {
-        return (new pristineEnv().Array(0)).concat(thing);
+    if (pristineObject.prototype.toString.call(thing) === '[object Array]') {
+        return (new pristineArray()).concat(thing);
     } else {
         return thing;
     }
 }
 
+/**
+ * Restore native implementations permanently.
```

Comment:
Might be worth mentioning that this will modify various prototypes by switching them back to their native implementation.